extensions/blocks/eventbrite/edit.js

```diff
@@ -38,106 +40,78 @@ class EventbriteEdit extends Component {
 	state = {
 		editedUrl: this.props.attributes.url || '',
 		editingUrl: false,
-		// Resolve the url on mount if we haven't already set an eventId,
-		// Such as when transforming from an Eventbrite link.
-		resolvingUrl: this.props.attributes.url && ! this.props.attributes.eventId,
-		resolvedStatusCode: null,
+		isResolvingUrl: false,
 	};
 
 	componentDidMount() {
-		const { resolvingUrl } = this.state;
+		const { url } = this.props.attributes;
 
-		// Check if we need to resolve an Eventbrite URL immediately.
-		if ( resolvingUrl ) {
-			this.resolveUrl();
-		}
+		this.setUrl( url );
 	}
 
-	componentDidUpdate( prevProps, prevState ) {
-		// Check if an Eventbrite URL has been entered, so we need to resolve it.
-		if ( ! prevState.resolvingUrl && this.state.resolvingUrl ) {
-			this.resolveUrl();
-		}
-	}
+	setUrl = url => {
+		const { attributes, noticeOperations, setAttributes } = this.props;
+		const { style } = attributes;
 
-	// TODO: figure out how to cancel request since apiFetch was updated to use Promises rather than XHR requests.
-	// componentWillUnmount() {}
+		if ( ! url || EVENTBRITE_EXAMPLE_URL === url || 'modal' === style ) {
+			return;
+		}
 
-	resolveUrl = () => {
-		const { url } = this.props.attributes;
+		const newAttributes = {
+			eventId: eventIdFromUrl( url ),
+			url,
+		};
 
-		this.setState( { resolvedStatusCode: null } );
+		testEmbedUrl( newAttributes.url, this.setIsResolvingUrl )
```

Comment:
for a follow up PR at this stage, but I wonder if we convert this to a functional component if we can abstract this further with a custom hook?

Reply:
Oh, that's interesting, I've never created a custom hook, so I didn't think of doing something like that!